dev/ansible/04_test_persistence.yml

```diff
@@ -0,0 +1,208 @@
+  register: nomad_postgres_status
+  when: lookup('env', 'nomad_acl') | bool
+  failed_when: nomad_postgres_status.stdout == "running"
+  environment:
+    NOMAD_TOKEN: "{{ bootstrap_token.stdout }}"
+  tags: test
+
+- name: Fetch nomad postgres job status (nomad acl disabled)
+  shell:
+    cmd: nomad status postgres | grep -m 1 'Status' | tr -d " \t\n\r"  | cut -d "=" -f2
+  register: nomad_postgres_status_1
+  when: not lookup('env', 'nomad_acl') | bool
+  failed_when: nomad_postgres_status_1.stdout == "running"
+  tags: test
+
+- name: Delete terraform related files and folders
+  file:
+    path: "../../example/{{ line_item }}"
+    state: absent
+  loop:
+    - terraform.tfstate.backup
+    - terraform.tfstate
+    - .terraform
+  loop_control:
+    loop_var: line_item
+  tags: test
+
+- name: Terraform apply to restart purged postgres job
+  terraform:
+    project_path: ../../example
+    force_init: true
+    state: present
+  register: terraform
+  tags: test
+
+- name: Terraform stdout
+  debug:
+    msg: "{{terraform.stdout}}"
+  tags: test
+
+- name: Verify that table exist by inserting data into table
+  shell: |
+    psql "dbname=metastore host=127.0.0.1 user={{ username.stdout }} password={{ password.stdout }} port=9999 sslmode=disable" < db_testfiles/data.sql
+  register: insert_data_output
+  retries: 5
+  delay: 15
+  failed_when: insert_data_output.stderr != ""
+  tags: test
+
+- name: Debug insert_data_output
+  debug:
+      msg: "{{ insert_data_output.stdout }}"
+  tags: test
+
+#****************************************#
+#Test Scenario 2
+#****************************************#
+# 1.Simulate container crash (docker cli, stop container and remove it)
+# 2.Nomad should automatically create a new instance of container with the same volume
+# 3.Verify that data available on new instance
+
+- name: Stop the docker container running postgres
+  shell: docker stop $(docker ps -a -q --filter ancestor="postgres:12-alpine")
+  register: docker_output
+  tags: test
+
+- name: Wait till nomad restarts the docker container running postgres
+  shell: docker ps -a -q --filter ancestor="postgres:12-alpine"
+  register: docker_output1
+  retries: 20
+  delay: 15
+  until: docker_output1.stdout != ""
+  tags: test
+
+- name: Start local proxy on port 9999 to postgres service
+  shell: nohup consul connect proxy -service postgres-local -upstream postgres:9999 -log-level debug </dev/null >/dev/null 2>&1 &
+  when: not lookup('env', 'consul_acl') | bool or (lookup('env', 'consul_acl') | bool and lookup('env', 'consul_acl_default_policy') == 'allow')
+  tags: test
+
+- name: Start local proxy on port 9999 to postgres service with token
+  shell: nohup consul connect proxy -service postgres-local -upstream postgres:9999 -log-level debug </dev/null >/dev/null 2>&1 &
+  when: lookup('env', 'consul_acl') | bool and lookup('env', 'consul_acl_default_policy') == 'deny'
+  environment:
+    CONSUL_HTTP_TOKEN: "{{ consul_token }}"
+  tags: test
```

Comment:
We might need to research that, proxies have already been running (at the beginning of the current playbook). For sure 2 out of 4 proxies are failing. I assume that these lines 184-194 are redundant, may be deleted due to lines 43-58. I need to verify that first

Reply:
<img width="1653" alt="Screenshot 2020-10-07 at 15 49 00" src="https://user-images.githubusercontent.com/15572799/95340500-877ab200-08b5-11eb-898f-9b2c448739da.png">
<img width="1760" alt="Screenshot 2020-10-07 at 15 54 48" src="https://user-images.githubusercontent.com/15572799/95340513-8b0e3900-08b5-11eb-849c-8b7766e4e272.png">

Reply:
Confirmed.